### PR TITLE
Add electron-builder workflow

### LINKED ISSUE #585: Add GitHub Action to build Electron app for all platforms
### Goal

Create a CI/CD script that builds our Electron application for all major platforms.

### Context

Currently, our Electron `build` script only builds for the host platform. We would like to extend that by building for all platforms via a GitHub Action.

### Assets

- `electron-builder` docs: https://www.electron.build/
- GitHub Actions docs: https://docs.github.com/en/actions

### Tasks

- [ ] Research how to add a CI/CD script to build for all platforms
- [ ] Implement it

### PR DESCRIPTION
**Pull request by aomafarag**
Closes #585.

Checklist:
- [x] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [x] issue checkboxes are all addressed
- [x] manually checked my feature / not applicable
- [x] wrote tests / not applicable
- [x] attached screenshots / not applicable

### CODE REVIEW

File: .github/workflows/build-electron-app.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Setup Node.js and NPM
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16
+
+      - name: Run electron-builder
+        uses: samuelmeuli/action-electron-builder@v1
+        with:
+          package_root: electron
+          github_token: ${{ secrets.github_token }}
```
[the remark this answers is not in the document]
Comment (valiafetisov) [reply .github/workflows/build-electron-app.yaml:28]: But my question was `why`? Can you provide link to the documentation that explains why it's needed, any other source or at least exact error which is thrown?

File: .github/workflows/build-electron-app.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    needs: build
+
+    permissions:
+      contents: write
+
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
```
Comment (valiafetisov) [inline .github/workflows/build-electron-app.yaml:50]: Does this job needs to run on all three platforms? I thought it only publishes the files, build is already done

File: .github/workflows/build-electron-app.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (valiafetisov) [reply .github/workflows/build-electron-app.yaml:28]: If this action gets `github_token` it's the same as giving write permissions (but maybe even more than that). Then, I don't actually see the need to split this process into 3 jobs and see the need to pin version of this action (via the hash) as well

**Review by valiafetisov — DISMISSED** [review]

**Comment by LukSteib** [comment]
Can you give me a quick explainer how I am supposed to test this?

**Comment by valiafetisov** [comment]
> Can you give me a quick explainer how I am supposed to test this?

I've thought that you have already tested previously. I think it should be possible by
1. Forking this repo into your private github account
2. Merging this PR into main
3. Making an empty release
4. Waiting for the CI to finish and add files to the release by editing it

[(not sure if you need to have paid account for this to work)](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes)

**Review by LukSteib — CHANGES_REQUESTED** [review]
For review I did the following steps: 
* forked this repo
* triggered a release from the `electron-builder-workflow` branch

- [ ] The Git CI ran into problems. Posting screenshots and errors below

image
image

**Comment by LukSteib** [comment]
> Please copy the [file from my fork](https://github.com/aomafarag/unified-auctions-ui/blob/main/electron/package.json), making sure to change repository.url to your fork's URL.

This means I'd have to open a PR for my fork and update the `package.json` accordingly, right?

**Review by LukSteib — APPROVED** [review]
Approving this. Was able to test release on fork

image

image

**Review by valiafetisov — APPROVED** [review]

**Pull request merged**